civicrm_entity.module

```diff
@@ -179,3 +179,66 @@ function _civicrm_entity_get_entity_type_from_elements($elements) {
 
   return NULL;
 }
+
+/**
+ * Implements hook_civicrm_pre().
+ */
+function civicrm_entity_civicrm_pre($op, $objectName, $id, &$params) {
+  $operations = ['create', 'update', 'delete'];
+  if (!in_array($op, $operations) || !$entityType = SupportedEntities::getDrupalType($objectName)) {
```

Comment:
I don't know if this actually violates Drupal style or not, but this bit seems ambiguous for humans: `!$entityType = SupportedEntities::getDrupalType($objectName)`

I think it'd be better to wrap the assignment in parenthesis and negate that.